WcaOnRails/app/models/competition.rb

```diff
@@ -156,6 +156,59 @@ def user_cannot_demote_themself
     end
   end
 
+  attr_accessor :receive_registration_emails
+  def receive_registration_emails=(r)
+    @receive_registration_emails = ActiveRecord::Type::Boolean.new.type_cast_from_database(r)
+  end
+
+  def receiving_registration_emails?(user_id)
+    competition_delegate = competition_delegates.find_by_delegate_id(user_id)
+    if competition_delegate && competition_delegate.receive_registration_emails
+      return true
+    end
+    competition_organizer = competition_organizers.find_by_organizer_id(user_id)
+    if competition_organizer && competition_organizer.receive_registration_emails
+      return true
+    end
+
+    return false
+  end
+
+  def can_receive_registration_emails?(user_id)
+    competition_delegate = competition_delegates.find_by_delegate_id(user_id)
+    if competition_delegate
+      return true
+    end
+    competition_organizer = competition_organizers.find_by_organizer_id(user_id)
+    if competition_organizer
+      return true
+    end
+
+    return false
+  end
+
+  validate :must_be_an_organizer_or_delegate_to_set_receive_registration_emails
+  def must_be_an_organizer_or_delegate_to_set_receive_registration_emails
```

Comment:
Could you move this check to the controller: https://github.com/jfly/worldcubeassociation.org/blob/master/WcaOnRails/app/controllers/competitions_controller.rb#L223 and simply remove the attribute from the permitted params?